tests/common/docker_image.py

```diff
@@ -0,0 +1,293 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains testing utilities."""
+import asyncio
+import logging
+import os
+import sys
+import time
+from abc import ABC, abstractmethod
+from threading import Timer
+from typing import Dict, Optional
+
+import docker
+import pytest
+import requests
+from docker import DockerClient
+from docker.models.containers import Container
+from oef.agents import OEFAgent
+from oef.core import AsyncioCore
+
+from aea.exceptions import enforce
+
+
+logger = logging.getLogger(__name__)
+
+
+class DockerImage(ABC):
+    """A class to wrap interatction with a Docker image."""
+
+    def __init__(self, client: docker.DockerClient):
+        """Initialize."""
+        self._client = client
```

Comment:
Problem is, we initiate before we skip. I think that won't work as docker won't be present on the platforms where we want to skip.

Reply:
Yes!! You're right